rest-api/src/main/java/gov/cms/qpp/conversion/api/model/Metadata.java

```diff
@@ -393,6 +405,49 @@ public void setCpcProcessed(Boolean cpcProcessed) {
 		this.cpcProcessed = cpcProcessed;
 	}
 
+	/**
+	 * Returns an attribute that combines the CPC+ processed state and the date of creation.
+	 *
+	 * This is mostly useful in the CPC+ global secondary index.
+	 *
+	 * @return The combined attribute.
+	 */
+	@DoNotEncrypt
+	@DynamoDBAttribute(attributeName = Constants.DYNAMO_CPC_PROCESSED_CREATE_DATE_ATTRIBUTE)
+	public String getCpcProcessedCreateDate() {
+		String combination = null;
+
+		if (cpcProcessed != null) {
+			combination = cpcProcessed.toString() + "#" + DateTimeFormatter.ISO_INSTANT.format(createdDate.toInstant());
+		}
+
+		return combination;
+	}
+
+	/**
+	 * Sets the separate CPC+ processed flag and created date based on the argument
+	 *
+	 * Splits the the processed flag from the date by a {@code #} character.
+	 * The first field must be {@code true} or {@code false} which represents the CPC+ processed boolean.
+	 * The second field must be an ISO 8601 timestamp string.  For example, {@code 2017-12-08T18:32:54.846Z}.
+	 *
+	 * @param combination The combined attribute.
+	 */
+	public void setCpcProcessedCreateDate(String combination) {
+
+		String[] split = combination.split("#");
+
+		if (split.length < CPC_PROCESSED_CREATE_DATE_NUM_FIELDS) {
+			return;
+		}
+
+		String isProcessed = split[CPC_PROCESSED_CREATE_DATE_NUM_FIELDS - 2];
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 2 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS109)